File: app/src/main/java/com/rr/razasypelajes/Horses/Horse.java
package com.rr.razasypelajes.Horses;

import android.support.annotation.NonNull;

import com.rr.razasypelajes.Helpers.JSONBuildable;

import org.json.JSONException;
import org.json.JSONObject;

public class Horse extends JSONBuildable {
    private String raza;
    private String pelaje;
    private String img;

    public Horse(JSONObject json) {
        super(json);
    }

    public String getRaza() {
        return raza;
    }

    public String getPelaje() {
        return pelaje;
    }

    public String getImg() {
        return img;
    }

    @NonNull
    @Override
    public String toString() {
        return "Horse{" +
                "raza='" + raza + '\'' +
                ", pelaje='" + pelaje + '\'' +
                ", img='" + img + '\'' +
                '}';
    }

    protected void fromJSON(JSONObject json) throws JSONException {
        this.raza = json.getString("raza");
        this.pelaje = json.getString("pelaje");
        this.img = json.getString("image");
    }
}

File: app/src/main/java/com/rr/razasypelajes/horses/Horse.kt
package com.rr.razasypelajes.horses

import android.content.Context
import com.rr.razasypelajes.helpers.JSONBuildable
import com.rr.razasypelajes.helpers.JSONHelper

import org.json.JSONException
import org.json.JSONObject

class Horse(json: JSONObject) : AbsHorse(json) {
    var raza: String? = null
        private set
    var pelaje: String? = null
        private set


    @Throws(JSONException::class)
    override fun fromJSON(json: JSONObject) {
        this.raza = json.getString("raza")
        this.pelaje = json.getString("pelaje")
        this.img = json.getString("image")
        this.name = json.getString("name")
        this.text = JSONHelper.stringFromJsonArray(json.getJSONArray("text"))
    }

    fun prettyRaza() : String? {
        return raza?.replace("_", " ")?.capitalize()
    }

    fun prettyPelaje() : String? {
        return pelaje?.replace("_", " ")?.capitalize()
    }

    fun prettyBoth() : String {
        return prettyRaza() + " - " + prettyPelaje()
    }

    override fun toString(): String {
        return "Horse(raza=$raza, pelaje=$pelaje, img=$img, name=$name, text=$text)"
    }

    override fun getReconSound(): String? {
        return raza + "_" + pelaje
    }

    override fun getReconText(): String? {
        return prettyBoth()
    }
}
